internal/handlers/common/saslstart.go

```diff
@@ -15,54 +15,44 @@
 package common
 
 import (
-	"bytes"
-	"encoding/base64"
+	"context"
 	"fmt"
 
+	"github.com/FerretDB/FerretDB/internal/clientconn/conninfo"
 	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
 	"github.com/FerretDB/FerretDB/internal/types"
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 )
 
-// SASLStartPlain extracts username and password from PLAIN `saslStart` payload.
-func SASLStartPlain(doc *types.Document) (string, string, error) {
-	var payload []byte
-
-	// some drivers send payload as a string
-	stringPayload, err := GetRequiredParam[string](doc, "payload")
-	if err == nil {
-		if payload, err = base64.StdEncoding.DecodeString(stringPayload); err != nil {
-			return "", "", lazyerrors.Error(err)
-		}
+// SASLStart returns a common part of saslStart command response.
+func SASLStart(ctx context.Context, doc *types.Document) error {
+	mechanism, err := GetRequiredParam[string](doc, "mechanism")
+	if err != nil {
+		return lazyerrors.Error(err)
 	}
 
-	// most drivers follow spec and send payload as a binary
-	binaryPayload, err := GetRequiredParam[types.Binary](doc, "payload")
-	if err == nil {
-		payload = binaryPayload.B
-	}
+	var username, password string
+
+	switch mechanism {
+	case "PLAIN":
+		username, password, err = saslStartPlain(doc)
```

Comment:
Should we also check and return error here instead of outside the switch?

Reply:
fixed